src/main/java/net/glowstone/net/message/play/game/PlayParticleMessage.java

```diff
@@ -10,9 +10,9 @@
     private final boolean longDistance;
     private final float x, y, z;
     private final float ofsX, ofsY, ofsZ;
-    private final float data;
+    private final float extData;
```

Comment:
Spigot API says that it's the other way around and I agree. Data is normally speed and can be used on all particle types unlike material data.